src/utils.ts

```diff
@@ -182,25 +188,62 @@ export const bufferToBase64UrlEncoded = input => {
   );
 };
 
-const getJSON = async (url, options) => {
-  const response = await fetch(url, options);
+const fetchWithTimeout = (url, options, timeout = DEFAULT_FETCH_TIMEOUT_MS) => {
+  // The promise will resolve with one of these two promises (the fetch and the timeout), whichever completes first.
+  return Promise.race([
+    fetch(url, options),
+    new Promise((_, reject) => {
```

Comment:
I'm sure this works for rejecting the promise early and eventually retrying. But to me, it feels like the `timeout` should be a feature/option of the node-fetch library. Because, if timeout kicks in, the request should be correctly canceled and that's only something the client can do.

Reply:
You are perhaps correct. However, we don't use `node-fetch` here, it's just the browsers Fetch API (polyfilled using [`promise-polyfill`](https://www.npmjs.com/package/promise-polyfill) and that has no protocol whatsoever for requests timing out. It will wait indefinitely for a response or until the connection drops.

Reply:
Then, do we get a way to cancel that `fetch(url, options)` promise gracefully?

Reply:
Yes, I've done some research into this and we can cancel `fetch` through the use of [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController). However, it has [no support on IE11](https://caniuse.com/#feat=abortcontroller) and thus would require a [polyfill](https://www.npmjs.com/package/abortcontroller-polyfill) and would bump the size of the package.

Let me investigate how big that bump would be and come back to you.

Reply:
@lbalmaceda I've implemented `AbortController` now and added a polyfill for IE11.

The polyfill bumps the distributed package size up to 16.55kb, a 12.4% increase.

Reply:
That doesn't seem like a huge increment, but I'd leave that to your discretion. Polyfill Q: Is there any way of only importing it on IE or is this something that if added, must be added in a general way?

Reply:
As it's all bundled, we have to add it in a generic way.